inspirehep/modules/workflows/views.py

```diff
@@ -376,3 +401,180 @@ def robotupload_callback():
         responses[recid] = response
 
     return jsonify(responses)
+
+
+def _validate_workflow_payload(keys):
+    """Validate and get the workflow data from the request.
+
+    It will first check if all the given keys are part of
+    the request's payload, after will validate the ``metadata`` against
+    the ``hep`` JSONSchema and finally it will pass the ``workflow_data``
+    kwarg.
+
+    Args:
+        keys (list(str)): the keys to check if they exist.
+
+    Raises:
+        CallbackMalformedError: if any of the ``keys`` is missing from
+            the request payload.
+        CallbackValidationError: if the workflow ``metadata`` is not valid
+            against ``hep`` JSONSchema.
+    """
+    workflow_data = request.get_json()
+
+    # Check for missing keys
+    for key in keys:
+        if workflow_data.get(key) is None:
+            raise CallbackMalformedError(key)
```

Comment:
This should also check that workflow_data is a dict-like object, and has a 'metadata' key.

Reply:
no, something else is off :/, looking

Reply:
The problem is that `workflow_data['_extra_data']` should be a dict, that check should be made here too and rise the `CallbackMalformedError` one.

Reply:
Passing an empty object as "id" also breaks with 500:
```
curl -vvv -H "content-type: application/json" -X PUT http://localhost:5000/callback/workflows/resolve_validation_errors -d '{"metadata": {"document_type": ["book"], "_collections": ["Literature"], "titles": [{"title": "snth"}]}, "_extra_data": {}, "id": {}}'
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 5000 (#0)
> PUT /callback/workflows/resolve_validation_errors HTTP/1.1
> Host: localhost:5000
> User-Agent: curl/7.55.1
> Accept: */*
> content-type: application/json
> Content-Length: 133
> 
* upload completely sent off: 133 out of 133 bytes
< HTTP/1.1 500 INTERNAL SERVER ERROR
< Server: gunicorn/19.7.1
< Date: Mon, 12 Mar 2018 18:59:05 GMT
< Connection: close
< Transfer-Encoding: chunked
< Content-Type: text/html; charset=utf-8
< X-XSS-Protection: 0
< 
```

Reply:
The proper way to handle this is marshmallow, which means we have to do needs refactoring to add loaders and schema. I'll do it for only this endpoint as a prototype.

Reply:
:+1: